omf/fileio/geoh5.py

```diff
@@ -0,0 +1,234 @@
+
+    def process_dependents(self, workspace):
+        ...
+
+    @abstractmethod
+    def from_geoh5(self) -> UidModel:
+        """Convert geoh5 entity to omg element."""
+        ...
+
+
+class PointsConversion(BaseConversion):
+    """"""
+    omf_type = PointSetElement
+    geoh5_type = Points
+    _attribute_map = {
+        "description": "description",
+        "name": "name",
+        "uid": "uid",
+        "geometry": "vertices",
+    }
+
+    def from_geoh5(self) -> PointSetElement:
+        """Convert geoh5 entity to omg element."""
+        ...
+
+    def process_dependents(self, workspace):
+        if getattr(self.element, "data", None):
+            for child in self.element.data:
+                converter = MAP[type(child)](child, workspace)
+                converter.from_omf(parent=self.entity)
+
+
+class VerticesConversion(BaseConversion):
+    """"""
+    omf_type = PointSetGeometry
+    geoh5_type = ndarray
+    _attribute_map = {}
+
+    def from_omf(self, **kwargs) -> ndarray:
+        return c_[self.element.vertices]
+
+    def from_geoh5(self) -> UidModel:
+        ...
+
+
+class DataConversion(BaseConversion):
+    """"""
+    omf_type = ScalarData
+    geoh5_type = Data
+    _attribute_map = {
+        "uid": "uid",
+        "name": "name",
+        "array": "values",
+        "colormap": "color_map"
+    }
+
+    def from_omf(self, parent=None, **kwargs):
+        with fetch_h5_handle(self.geoh5) as workspace:
+
+            for key, alias in self._attribute_map.items():
+                prop = getattr(self.element, key, None)
+                if isinstance(prop, UidModel):
+                    converter = MAP[type(prop)](prop, workspace)
+                    prop = converter.from_omf()
+
+                kwargs[alias] = prop
+
+            self._entity = parent.add_data({self.element.name: kwargs})
+
+    def from_geoh5(self) -> UidModel:
+        ...
+
+
+class ValuesConversion(BaseConversion):
+    omf_type = ScalarArray
+    geoh5_type = ndarray
+    _attribute_map = {}
+
+    def from_omf(self, parent=None, **kwargs) -> ndarray:
+        return r_[self.element.array]
+
+    def from_geoh5(self) -> UidModel:
+        ...
+
+
+@contextmanager
+def fetch_h5_handle(file: str | Workspace | Path, mode: str = "a") -> Workspace:
+    """
+    Open in read+ mode a geoh5 file from string.
+    If receiving a file instead of a string, merely return the given file.
+
+    :param file: Name or handle to a geoh5 file.
+    :param mode: Set the h5 read/write mode
+
+    :return h5py.File: Handle to an opened h5py file.
+    """
+    if isinstance(file, Workspace):
+        try:
+            yield file
+        finally:
+            pass
+    else:
+        if Path(file).suffix != ".geoh5":
+            raise ValueError("Input h5 file must have a 'geoh5' extension.")
+
+        h5file = Workspace(file, mode)
+
+        try:
+            yield h5file
+        finally:
+            h5file.close()
+
+
+MAP = {
```

Comment:
maybe an internal variable, not to be used by client code? `_MAP`

Reply:
Done